airflow/www/views.py

```diff
@@ -92,7 +93,10 @@ def get_date_time_num_runs_dag_runs_form_data(request, session, dag):
     if dttm:
         dttm = pendulum.parse(dttm)
     else:
-        dttm = dag.latest_execution_date or timezone.utcnow()
+        execution_date = session.query(sqla.func.max(DagRun.execution_date)).filter(
+            DagRun.dag_id == dag.dag_id
+        ).scalar()
```

Comment:
Why have you expanded this? Isn't that exactly what `dag.last_execution_date` does? (and keeps DB access logic out of the views which makes me happier)

Reply:
This should be temporary code. The problem here is that right now in views.py DAG and DagModel is used both for this method. `dag_id` is at least something both classes has in common. Accessing a method will break this. Once all is switched from DAG to DagModel this can be fixed again.